app/Traits/JsonResponseTrait.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Traits;
+
+use Illuminate\Http\JsonResponse;
+
+trait JsonResponseTrait
+{
+
+    /**
+     * Generate a JSON response for successful operations.
+     *
+     * @param  mixed  $data
+     * @param  string  $messageKey
+     * @param  int  $statusCode
+     * @return JsonResponse
+     */
+    public function successResponse($data, $messageKey = 'success', $statusCode = 200): JsonResponse
+    {
+        return response()->json([
+            'success' => true,
+            'message' => trans($messageKey),
+            "message_code"=> $messageKey,
+            'data' => $data,
+        ], $statusCode);
+    }
+
+    /**
+     * Generate a JSON response for errors.
+     *
+     * @param  string  $message
+     * @param  string  $messageKey
+     * @param  int  $statusCode
+     * @return JsonResponse
+     */
+    public function errorResponse( $messageKey = 'error', $statusCode = 500): JsonResponse
+    {
+        return response()->json([
```

Comment:
This funciton should return the errordata also if I pass 
If I have errors with validaiton where I will pass my error data?

Reply:
resolve